backend-config/replay_types.go

```diff
@@ -0,0 +1,85 @@
+package backendconfig
+
+import (
+	"github.com/samber/lo"
+)
+
+type EventReplayConfigs map[string]*EventReplayConfig
+
+// ApplyReplaySources reads the event replay configuration and adds replay sources to the config
+// A replay source is a copy of the original source with a different ID and source definition
+// This replay source contains as destinations replay destinations which are copies of the original destinations but with a different ID
+func (config *ConfigT) ApplyReplaySources() {
+	if len(config.EventReplays) == 0 {
+		return
+	}
+	originalSources := config.SourcesMap()
+	originalDestinations := config.DestinationsMap()
+	for _, replay := range config.EventReplays {
+		sources := lo.OmitByValues(lo.MapValues(replay.Sources, func(value EventReplaySource, id string) *SourceT {
+			s, ok := originalSources[value.OriginalSourceID]
+			if !ok {
+				return nil
+			}
+			newSource := *s
+			newSource.ID = id
+			newSource.WriteKey = id
+			newSource.EventSchemasEnabled = false
+			newSource.SourceDefinition = SourceDefinitionT{
```

Comment:
If existence of a source definition in config backend is imperative for reporting to function properly (or for any other component on the same matter), yes. Otherwise we can keep this source definition internal to rudder-server.

Reply:
[This](https://github.com/rudderlabs/rudderstack-reporting/blob/master/src/api/producer/v0.ts#L85) is the code flow in reporting

Reply:
So by default, if the id doesn't exist in source definitions, it will be interpreted as [EventStream](https://github.com/rudderlabs/rudderstack-reporting/blob/master/src/api/producer/v0.ts#L99) by reporting?

Reply:
leaving the pull request open for the time being, until we sort this out

Reply:
For the time being, replay sources are going to use the same source definition as their original sources